community/values/src/main/java/org/neo4j/values/storable/DateTimeValue.java

```diff
@@ -482,17 +501,24 @@ int unsafeCompareTo( Value other )
             cmp = value.toLocalTime().getNano() - that.value.toLocalTime().getNano();
             if ( cmp == 0 )
             {
-                cmp = value.toLocalDateTime().compareTo( that.value.toLocalDateTime() );
+                ZoneOffset thisOffset = value.getOffset();
+                ZoneOffset thatOffset = that.value.getOffset();
+
+                cmp = Integer.compare( thisOffset.getTotalSeconds(), thatOffset.getTotalSeconds() );
                 if ( cmp == 0 )
                 {
                     ZoneId thisZone = value.getZone();
                     ZoneId thatZone = that.value.getZone();
                     boolean thisIsOffset = thisZone instanceof ZoneOffset;
                     boolean thatIsOffset = thatZone instanceof ZoneOffset;
-                    cmp = Boolean.compare( thatIsOffset, thisIsOffset ); // offsets before named zones
-                    if ( cmp == 0 && (thisIsOffset || !areSameZone( thisZone, thatZone, thisIsOffset, thatIsOffset ) ) )
+                    // non-named timezone (just offset) before named-time zones, alphabetically
```

Comment:
The comment here says to order by zones alphabetically, which I assume means alphabetically by name...